src/Symfony/Bundle/SecurityBundle/Tests/Functional/AutowiringTypesTest.php

```diff
@@ -22,6 +22,15 @@ public function testAccessDecisionManagerAutowiring()
         $this->assertInstanceOf('Symfony\Component\Security\Core\Authorization\AccessDecisionManager', $accessDecisionManager);
     }
 
+    public function testDebugAccessDecisionManagerAutowiring()
+    {
+        static::bootKernel(array('debug' => true));
+        $container = static::$kernel->getContainer();
+
+        $accessDecisionManager = $container->get('test.autowiring_types.autowired_services')->getAccessDecisionManager();
+        $this->assertInstanceOf('Symfony\Component\Security\Core\Authorization\AccessDecisionManager', $accessDecisionManager);
```

Comment:
Ah I thought this was a feature, nevertheless, good to have this test so we will notice it breaks in 3.1 and it can be fixed there.

I think it's correct that autowiring are allowed in lower versions, but I'm not sure. Following semver it would not be allowed because if you rely on the functionality and you go back 1 patch due to a bug fix bug, you also break this "feature".